### PR TITLE
RAII temporary directory helper class

### PR DESCRIPTION
**Pull request by radupopescu**
Addresses [CVM-1306](https://sft.its.cern.ch/jira/browse/CVM-1306).

Provides a helper class to create scoped temporary directories - the directories are created in the constructor of the object and removed in its destructor.

Some classes used in the `cvmfs_receiver` application have beem modified to use this class, to avoid leaving rubbish files behind.

### CODE REVIEW

File: test/unittests/t_raii_temp_dir.cc [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <gtest/gtest.h>
+
+#include <util/pointer.h>
```
Comment (jblomer) [inline test/unittests/t_raii_temp_dir.cc]: Should be "util/pointer.h"?

File: test/unittests/t_raii_temp_dir.cc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <util/raii_temp_dir.h>
```
Comment (jblomer) [inline test/unittests/t_raii_temp_dir.cc]: "util/raii_temp_dir.h"

File: test/unittests/t_raii_temp_dir.cc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+static bool DirExists(const std::string& dir) {
+  platform_stat64 dir_stat;
+  int ret = platform_stat(dir.c_str(), &dir_stat);
+  return ret == 0;
+}
+
+class T_RaiiTempDir : public ::testing::Test {};
+
+TEST_F(T_RaiiTempDir, Basic) {
+  UniquePtr<RaiiTempDir> temp_dir(RaiiTempDir::Create("/tmp/test_dir"));
```
Comment (jblomer) [inline test/unittests/t_raii_temp_dir.cc]: The unit tests already have there working directory on a temporary path.  Creation of files and directories should start in `GetCurrentWorkingDirectory()`

**Comment by jblomer** [comment]
Sorry, I didn't have time to review this any earlier.

File: test/unittests/t_raii_temp_dir.cc (hunk 3 above)
Comment (radupopescu) [inline test/unittests/t_raii_temp_dir.cc]: Neat, I didn't know about that!

**Comment by radupopescu** [comment]
No worries!

**Pull request merged**